sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -255,6 +255,41 @@ case class HashAggregateExec(
      """.stripMargin
   }
 
+  // Splits aggregate code into small functions because the most of JVM implementations
+  // can not compile too long functions. Note that different from `CodeGenerator.splitExpressions`,
+  // we will extract input variables from references and subexpression elimination states
+  // for each aggregate expression, then pass them to it.
+  private def splitAggregateExpressions(
+      context: CodegenContext,
+      aggregateExpressions: Seq[Expression],
+      codes: Seq[String],
+      subExprs: Map[Expression, SubExprEliminationState],
+      bufferInput: Option[(String, DataType)] = None): Seq[String] = {
```

Comment:
Same question on `(String, DataType)` being just a degenerated `VariableValue`

Reply:
Ur, I see. I replaced it with VariableValue. Thanks!